### PR TITLE
feat(bazel): add "maybe" macro support

### PR DESCRIPTION
**Pull request by jbedard**
## Changes

Adding support for the [maybe](https://docs.bazel.build/versions/0.29.1/repo/utils.html#maybe) function.

## Context

This is a util function recommended instead of always wrapping imports in `if`.

IMO there's probably a better way of doing this by making the lexer more specific/realistic instead of just matching any type of import as a giant string. That's probably a bigger change for another time though...

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by jbedard** [comment]
Is there an easy way of testing this in another repo? I'd like to test it in a real environment and not just the unit tests...

FYI @alexeagle

**Comment by alexeagle** [comment]
Some add'l background for maintainers here: the "maybe" function is defined here:
https://github.com/bazelbuild/bazel/blob/6530488ba1914645b1e24f9c6f132ecf5f8a2ccb/tools/build_defs/repo/utils.bzl#L218-L233

It can be pretty trivially inlined, so the workaround for all Bazel users until we have this feature is to de-sugar it like:
https://github.com/aspect-build/rules_py/pull/60

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/bazel/extract.ts [hunk 1]
```diff
@@ -73,13 +73,16 @@ const lexer = moo.states({
...
+        '(?:' +
+          [
+            'container_pull',
+            'http_archive',
+            'http_file',
+            'go_repository',
+            'git_repository',
+            'maybe',
+          ].join('|') +
+          ')\\s*\\('
```
Comment (zharinov) [inline lib/modules/manager/bazel/extract.ts:85]: Parens should be handled somewhere else

```suggestion
        [
          'container_pull',
          'http_archive',
          'http_file',
          'go_repository',
          'git_repository',
          'maybe',
        ].join('|')
```

**Review by zharinov — CHANGES_REQUESTED** [review]

**Review by zharinov — DISMISSED** [review]

**Comment by viceice** [comment]
needs null check fix

**Review by rarkins — APPROVED** [review]

**Pull request merged**